### PR TITLE
Add Typo Tolerance settings

### PR DESCRIPTION
**Pull request by brunoocasali**
Add new methods to handle the typo tolerance settings from an Index setting.

### CODE REVIEW

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -373,3 +373,53 @@ tenant_token_guide_generate_sdk_1: |-
 tenant_token_guide_search_sdk_1: |-
   frontEndClient = new MeilisearchClient("http://127.0.0.1:7700", token);
   SearchResult<Patient> searchResult = await frontEndClient.Index("patient_medical_records").SearchAsync<Patient>("blood test");
+get_typo_tolerance_1:
```
Comment (alallema) [inline .code-samples.meilisearch.yaml:376]: The `UpdateSetting` method doesn't exist in `dotnet` right?

File: src/Meilisearch/Settings.cs [hunk 2]
```diff
@@ -55,5 +55,11 @@ public class Settings
         /// </summary>
         [JsonPropertyName("sortableAttributes")]
         public IEnumerable<string> SortableAttributes { get; set; }
+
+        /// <summary>
+        /// Gets or sets the typo tolerance configuration attributes.
```
Comment (alallema) [inline src/Meilisearch/Settings.cs]:
```suggestion
        /// Gets or sets the typo tolerance attributes.
```
Not sure about that but it seems to be more understandable without configuration for me.

File: src/Meilisearch/TypoTolerance.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        /// <summary>
+        /// Customize the minimum size for a word to tolerate typos
+        /// </summary>
+        [JsonPropertyName("minWordSizeForTypos")]
+        public TypoSize MinWordSizeForTypos { get; set; }
+
+        public class TypoSize
+        {
+            /// <summary>
+            /// Customize the minimum size for a word to tolerate 1 typo.
```
Comment (alallema) [inline src/Meilisearch/TypoTolerance.cs]:
```suggestion
            /// Customize the minimum word size to tolerate 1 typo.
```

File: src/Meilisearch/TypoTolerance.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        public IEnumerable<string> DisableOnAttributes { get; set; }
+
+        /// <summary>
+        /// Disable the typo tolerance feature for a set of query terms given during a search query.
+        /// </summary>
+        [JsonPropertyName("disableOnWords")]
+        public IEnumerable<string> DisableOnWords { get; set; }
+
+        /// <summary>
+        /// Customize the minimum size for a word to tolerate typos
```
Comment (alallema) [inline src/Meilisearch/TypoTolerance.cs]:
```suggestion
        /// Customize the minimum word size to tolerate typos.
```

File: src/Meilisearch/TypoTolerance.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        public bool Enabled { get; set; }
+
+        /// <summary>
+        /// Disable the typo tolerance feature on the specified document attributes.
+        /// </summary>
+        [JsonPropertyName("disableOnAttributes")]
+        public IEnumerable<string> DisableOnAttributes { get; set; }
+
+        /// <summary>
+        /// Disable the typo tolerance feature for a set of query terms given during a search query.
```
Comment (alallema) [inline src/Meilisearch/TypoTolerance.cs]:
```suggestion
        /// Disable the typo tolerance feature for a given set of terms in a search query.
```

File: src/Meilisearch/TypoTolerance.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        public class TypoSize
+        {
+            /// <summary>
+            /// Customize the minimum size for a word to tolerate 1 typo.
+            /// </summary>
+            [JsonPropertyName("oneTypo")]
+            public int? OneTypo { get; set; }
+
+            /// <summary>
+            /// Customize the minimum size for a word to tolerate 2 typos.
```
Comment (alallema) [inline src/Meilisearch/TypoTolerance.cs]:
```suggestion
            /// Customize the minimum word size to tolerate 2 typos.
```

**Review by alallema — COMMENTED** [review]
Well Done!!! 🔥

File: .code-samples.meilisearch.yaml (hunk 1 above)
Comment (brunoocasali) [reply .code-samples.meilisearch.yaml:376]: I didn't understand your question, but I realized that I missed the suffix "Async" :)

**Review by alallema — APPROVED** [review]
LGTM! 🥖

**Pull request merged**